### PR TITLE
Only softfail with a state kwarg mismatch

### LINKED ISSUE #20227: Recent behavior in develop about state kwarg enforcement
```
[INFO    ] Executing state mc_proxy.hook for foo
Passed invalid arguments: 'bar' is an invalid keyword argument for 'mc_proxy.hook'. If you were trying to pass additional data to be used in a template context, please populate 'context' with 'key: value' pairs. Please update your state files.
Traceback (most recent call last):
  File "/srv/mastersalt/makina-states/src/salt/salt/cli/caller.py", line 139, in call
    ret['return'] = func(*args, **kwargs)
  File "/srv/mastersalt/makina-states/src/salt/salt/modules/state.py", line 640, in sls
    ret = st_.state.call_high(high_)
  File "/srv/mastersalt/makina-states/src/salt/salt/state.py", line 2047, in call_high
    ret = dict(list(disabled.items()) + list(self.call_chunks(chunks).items()))
  File "/srv/mastersalt/makina-states/src/salt/salt/state.py", line 1611, in call_chunks
    running = self.call_chunk(low, running, chunks)
  File "/srv/mastersalt/makina-states/src/salt/salt/state.py", line 1863, in call_chunk
    running[tag] = self.call(low, chunks, running)
  File "/srv/mastersalt/makina-states/src/salt/salt/state.py", line 1498, in call
    expected_extra_kws=STATE_INTERNAL_KEYWORDS
  File "/srv/mastersalt/makina-states/src/salt/salt/utils/__init__.py", line 801, in format_call
    '' if 'full' not in ret else ' Please update your state files.'
SaltInvocationError: 'bar' is an invalid keyword argument for 'mc_proxy.hook'. If you were trying to pass additional data to be used in a template context, please populate 'context' with 'key: value' pairs. Please update your state files.
```

As i understand the solution, i really feel it a bit too rude/much to totally fail here, a lot of states that we use change a bit overtime but are mostly compatible.

Shall we just log something (like with an error level) rather than failing ?
This is also problematic if we have duplicated a lot of states which ... contain 'non blocking' errors :D

cc @thatch45

### PR DESCRIPTION
**Pull request by kiorky**
This rels & fix #20227.

### CODE REVIEW

File: salt/utils/__init__.py [hunk 1]
```diff
@@ -793,7 +793,7 @@ def format_call(fun,
                     '{0}.{1}'.format(fun.__module__, fun.__name__)
                 )
             )
-        raise SaltInvocationError(
+        log.error(
```
Comment (kiorky) [inline salt/utils/__init__.py]: we can also increase to critical.
What are you thinking about ?

**Comment by thatch45** [comment]
Yes, I am ok with critical

**Comment by thatch45** [comment]
I should say that I am ok either way, I am in the fence so to speak

**Comment by kiorky** [comment]
increased loglevel, cc @thatch45

**Comment by s0undt3ch** [comment]
We've been returning warnings for some versions now. At most we postpone the deprecation a fee versions further down but the exception should be raised....

**Comment by s0undt3ch** [comment]
*few

**Comment by s0undt3ch** [comment]
Its the only way we can tell users that they are passing bad arguments....

We will start getting, I pass X argument and nothing happened, again...

**Pull request merged**